jsonrpc/server.go

```diff
@@ -359,20 +377,33 @@ func (s *Server) handleBatchRequest(ctx context.Context, batchReq []json.RawMess
 					resp.ID = req.ID
 				}
 			}
-			// for notification request response is nil
+			// for notification request response is nil and header is irrelevant for now
 			if resp != nil {
-				addResponse(resp)
+				addResponse(resp, header)
 			}
 		})
 	}
 
 	wg.Wait()
+
+	// merge headers
+	finalHeaders := http.Header{}
+	for _, header := range headers {
+		for k, v := range header {
+			for _, e := range v {
+				finalHeaders.Add(k, e)
+			}
+		}
+	}
```

Comment:
We are only looking to add a single header, so I don't think we need to account for multiple headers, imo